src/wfdlgs2.c

```diff
@@ -234,17 +236,25 @@ SearchDlgProc(register HWND hDlg, UINT wMsg, WPARAM wParam, LPARAM lParam)
                       // Forward the attributes to the search window, since hwndSearch was just created by WM_MDICREATE
                       SetWindowLongPtr(hwndSearch, GWL_ATTRIBS, GetWindowLongPtr(hwndMDIChild, GWL_ATTRIBS));
 
-                      if (bMaximized)
-                      {
-                          // the WM_MDICREATE above creates the window maximized;
+                      if (bMaximized) {
+                         // the WM_MDICREATE above creates the window maximized;
                           // here we re-activate the original maximized window
 
                           SendMessage(hwndMDIClient, WM_MDIACTIVATE, (WPARAM)hwndMDIChild, 0L);
 
                           SendMessage(hwndMDIClient, WM_SETREDRAW, TRUE, 0);
                           RedrawWindow(hwndMDIClient, NULL, NULL, RDW_INVALIDATE | RDW_ALLCHILDREN);
+                      } else {
+                         SendMessage(hwndSearch, FS_CHANGEDISPLAY, CD_PATH, (LPARAM)SearchInfo.szSearch);
                       }
                   }
+
+                  // Show search window immediatley
+                  ShowWindow(hwndSearch,
+                     GetWindowLongPtr(hwndMDIChild, GWL_STYLE) & WS_MAXIMIZE ?
+                     SW_SHOWMAXIMIZED :

```

Comment:
Seems to me this should use bMaximized, not fetch the style of the child.

Reply:
Changed with 503da52.
Sry for the confusion, but up till now I was not able to reproduce the bad-case, So I was in the dark.
Anyhow, I hope it is fine now and solves the 'bad use-case'.

Reply:
@craigwims: Is this one good to go?